### PR TITLE
Issue #11637: Resolve shellcheck SC2005

### PR DESCRIPTION
**Pull request by shubh220922**
Related to #11637 

https://www.shellcheck.net/wiki/SC2005
> Problematic code:
echo "$(whoami)"
Correct code:
whoami

### CODE REVIEW

**Comment by shubh220922** [comment]
@romani can you see if changes make sense. There are places where the command is in double quotes instead of ``(command)``
so I didn't know what to do with that:
example is as below:
``RELEASE_ID="$RELEASE_ID"``

I've run shellcheck on every bash file under ``/.ci/`` and it doesn't report SC2005 warning for any of the files after commiting the changes.

**Comment by shubh220922** [comment]
image

My shellcheck is not showing any warning for line 7. @romani

**Comment by rnveach** [comment]
@shubh220922 Please re-read first post since I updated it with the link and information on SC2005.

It says it is for things like `echo "$(command)"`. Some of the commands you changed do not follow that pattern. You will have to show us they are functionally the same if you believe the changes are valid.

File: .ci/util.sh [hunk 1]
```diff
@@ -16,9 +16,9 @@ function checkForVariable() {
 
 function getMavenProperty {
   property="\${$1}"
-  echo "$(mvn -e --no-transfer-progress -q -Dexec.executable='echo' \
+   mvn -e --no-transfer-progress -q -Dexec.executable='echo' \
```
Comment (rnveach) [inline .ci/util.sh:19]: ALso, indentation is incorrect. -1 space.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: .ci/util.sh (hunk 1 above)
Comment (shubh220922) [reply .ci/util.sh]: Done @rnveach .

**Comment by shubh220922** [comment]
@rnveach This time I have only changed only those lines that match exactly with the pattern mentioned in ``SC2005`` as below:
```disable=SC2005 # (style): Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.```

The below lines have been changed for the mentioned reasons:

```
  echo "$(mvn -e --no-transfer-progress -q -Dexec.executable='echo' \
                      -Dexec.args="${property}" \
                      --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)"
```
changed to
```
  mvn -e --no-transfer-progress -q -Dexec.executable='echo' \
                      -Dexec.args="${property}" \
                      --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec
```

but shellcheck says there is no problem with the file at all which means my changes were not required.

```
PS C:\Users\shubh\OneDrive\Desktop\checkstyle project\checkstyle\.ci> cat util.sh
#!/bin/bash
# This script contains common bash CI functions
set -e

removeFolderWithProtectedFiles() {
  find "$1" -delete
}

function checkForVariable() {
  VAR_NAME=$1
  if [ ! -v "$VAR_NAME" ]; then
    echo "Error: Define $1 environment variable"
    exit 1
  fi
}

function getMavenProperty {
  property="\${$1}"
  echo "$(mvn -e --no-transfer-progress -q -Dexec.executable='echo' \
                      -Dexec.args="${property}" \
                      --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)"
}

function getCheckstylePomVersion {
  getMavenProperty project.version
}

function checkout_from {
  CLONE_URL=$1
  PROJECT=$(echo "$CLONE_URL" | sed -nE 's/.*\/(.*).git/\1/p')
  mkdir -p .ci-temp
  cd .ci-temp
  if [ -d "$PROJECT" ]; then
    echo "Target project $PROJECT is already cloned, latest changes will be fetched and reset"
    cd "$PROJECT"
    git fetch
    git reset --hard HEAD
    git clean -f -d
    cd ../
  else
    for i in 1 2 3 4 5; do git clone "$CLONE_URL" && break || sleep 15s; done
  fi
  cd ../
}
PS C:\Users\shubh\OneDrive\Desktop\checkstyle project\checkstyle\.ci> shellcheck util.sh
PS C:\Users\shubh\OneDrive\Desktop\checkstyle project\checkstyle\.ci> 
```

**Comment by rnveach** [comment]
> shellcheck says there is no problem with the file at all which means my changes were not required.(CLI below)

You did not provide `git diff` or `git status` from your CLI execution so it is unclear what state your repo was in when you made the changes.

To show that shellcheck will print a violation on the file, you need to update this PR to only have the update at `.shellcheckrc` and nothing else. When you do that, provide us the link to the CI results of shellcheck.

**Comment by shubh220922** [comment]
@rnveach I did the changes you asked and made chagnes only in the ``.shellcheckrc`` file. I am linking the CI result below;
https://github.com/checkstyle/checkstyle/actions/runs/4064571196/jobs/6998164065#step:4:8

**Comment by shubh220922** [comment]
Sorry wanted to delete a comment, ended up closing the PR. Have reopened it.

**Comment by shubh220922** [comment]
@rnveach I did the changes you asked and made changes only in the ``.shellcheckrc`` file. I am linking the CI result below;
https://github.com/checkstyle/checkstyle/actions/runs/4064571196/jobs/6998164065#step:4:8

result of ``git diff`` is below:

```
$ git diff master shellcheck
diff --git a/.shellcheckrc b/.shellcheckrc
index 52bb45198..bb47f53c2 100644
--- a/.shellcheckrc
+++ b/.shellcheckrc
@@ -21,7 +21,6 @@ disable=SC2128 # (warning): Expanding an array without an index only gives the f
 disable=SC2063 # (warning): Grep uses regex, but this looks like a glob.
 disable=SC2126 # (style): Consider using grep -c instead of grep|wc -l.
 disable=SC2088 # (warning): Tilde does not expand in quotes. Use $HOME.
-disable=SC2005 # (style): Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
 disable=SC2015 # (info): Note that A && B || C is not if-then-else. C may run when A is true.
 disable=SC2026 # (info): This word is outside of quotes.
 disable=SC2230 # which is non-standard. Use builtin 'command -v' instead.

```

**Comment by rnveach** [comment]
@shubh220922 If your sure your commit is correct, then putting it back in will show no CI errors. Please do this and post the CI link of the run.

If you ever feel there is a false negative (an issue is not being reported by CI), then we will need to submit an issue to the owners of the tool to examine it further.

**Comment by shubh220922** [comment]
@rnveach last commit restored.
CI passing without issue.

CI that was previously failing (https://github.com/checkstyle/checkstyle/actions/runs/4064571196/jobs/6998164065#step:4:8) has been fixed:
https://github.com/checkstyle/checkstyle/actions/runs/4080885656/jobs/7033775262#step:4:2

**Comment by shubh220922** [comment]
@rnveach last commit restored.
CI passing without issue.

CI that was previously failing (https://github.com/checkstyle/checkstyle/actions/runs/4064571196/jobs/6998164065#step:4:8) has been fixed:
https://github.com/checkstyle/checkstyle/actions/runs/4080885656/jobs/7033775262#step:4:2

**Comment by rnveach** [comment]
Thanks. There is only 1 thing left then for approval from me.

https://github.com/checkstyle/checkstyle/pull/12688#issuecomment-1407761631
> Please also show us an example of running your change to show they are correct.

I want to ensure this check is correct and the output will still be displayed and the functionality of the script is not impacted.

**Comment by shubh220922** [comment]
> > Please also show us an example of running your change to show they are correct.
> 
> I want to ensure this check is correct and the output will still be displayed and the functionality of the script is not impacted.

@rnveach I am sorry but I don't understand how to do that. Do you want me to run the bash script through terminal?
It doesn't show any output though (I've run it through git bash):

```

shubh@ShubhLenovo7i MINGW64 ~/OneDrive/Desktop/checkstyle project/checkstyle/.ci (shellcheck)
$ bash util.sh

shubh@ShubhLenovo7i MINGW64 ~/OneDrive/Desktop/checkstyle project/checkstyle/.ci (shellcheck)
$
```

**Review by romani — APPROVED** [review]
Ok to merge

**Comment by romani** [comment]
@shubh220922, please rebase and resolve conflict

**Comment by rnveach** [comment]
@shubh220922 
> and the functionality of the script is not impacted

This was all I was looking for. If no output is displayed, then I just wanted confirmation that script will behave the same as before and after your change.

> $ bash util.sh

This is a function, and will not be executed normally through just the script. You will have to identify who calls and examine their script run.

It is run from https://github.com/checkstyle/checkstyle/blob/50e41f4aad67c028c6d422cdf3aec68240e12fd5/.ci/validation.sh#L260
And is assigned to the property `ANTLR4_VERSION`. So you could unset the variable, run the script without your change, echo the variable, and then redo the steps again with your change.

**Comment by shubh220922** [comment]
@rnveach 
> This is a function, and will not be executed normally through just the script. You will have to identify who calls and examine their script run.
> 
> It is run from
> 
> https://github.com/checkstyle/checkstyle/blob/50e41f4aad67c028c6d422cdf3aec68240e12fd5/.ci/validation.sh#L260
> 
> 
> And is assigned to the property `ANTLR4_VERSION`. So you could unset the variable, run the script without your change, echo the variable, and then redo the steps again with your change.

This is helpful information.

**Comment by shubh220922** [comment]
@romani @rnveach repushed after updating branch.

**Comment by shubh220922** [comment]
oops. CI failing:
https://github.com/checkstyle/checkstyle/actions/runs/4094103319/jobs/7060018316#step:6:336

fixing it.

**Comment by rnveach** [comment]
> util.sh is behaving normal:

Again, you can't run the script like that. You have to run validation.sh with the specific arguments to show the usage. Let me know if your not understanding my previous points.

**Comment by nrmancuso** [comment]
@shubh220922 a few things here:

- when you are trying to duplicate errors locally (for anything, not just shellcheck), make sure to use the exact same version and run it in the same manner (CLI)
- You CAN directly run a function from a bash script, but you must make sure that the contents of the script are in the context. The easiest way is probably something like `source  util.sh && getMvnVersion`

**Comment by shubh220922** [comment]
> It is run from
> 
> https://github.com/checkstyle/checkstyle/blob/50e41f4aad67c028c6d422cdf3aec68240e12fd5/.ci/validation.sh#L260
> 
> 
> And is assigned to the property `ANTLR4_VERSION`. So you could unset the variable, run the script without your change, echo the variable, and then redo the steps again with your change.

Okay now I totally get how these files a communicating with each other. 
Studying them deeply now.

**Comment by shubh220922** [comment]
@nrmancuso noted.
I have some doubts though.


> * when you are trying to duplicate errors locally (for anything, not just shellcheck), make sure to use the exact same version and run it in the same manner (CLI)

Here you asked me to use the exact same version whilst duplicating errors. Version of what are you referring to ?
Also what made you say this? In the sense what thing did i do wrong in this PR.

**Comment by nrmancuso** [comment]
> Here you asked me to use the exact same version whilst duplicating errors. Version of what are you referring to ?

I see screenshots in PR conversation from shellcheck IDE plug-in, it is better to run it via CLI using the exact same version that we use in CI.

**Comment by shubh220922** [comment]
```
PS C:\Users\shubh> shellcheck --version
ShellCheck - shell script analysis tool
version: 0.9.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
PS C:\Users\shubh>
```
@nrmancuso since you mentioned about version. I am using the above version. I read https://github.com/checkstyle/checkstyle/issues/11637 but nowhere any specific version was explicitely mentioned so I am working with the latest one available.

**Comment by shubh220922** [comment]
Progress of this PR has been resumed in .

**Comment by nrmancuso** [comment]
@shubh220922 please learn how to use git/GitHub so that you can avoid opening new PRs in the future. Existing conversations and context are left behind.

**Comment by shubh220922** [comment]
> @shubh220922 please learn how to use git/GitHub so that you can avoid opening new PRs in the future. Existing conversations and context are left behind.

Yes I totally understand.

I'll take care of it.

**Comment by nrmancuso** [comment]
@shubh220922 research git reflog, this is a powerful tool that can help you to fix lots of problems.

**Comment by shubh220922** [comment]
@nrmancuso thanks a lot. I remember when I started contributing here, I used git reflog but eventually forgot about it. 

I migrated from windows to linux(fedora 37) and that's why had a hard time with setup.

**Pull request closed without merging**